### DISCUSSION TITLE
Playing multiple audio files in a row using A2DP Source

### DISCUSSION THREAD

**Discussion by SamuHermi** [pschatzmann/ESP32-A2DP#208] [Q&A] [answered]
Hi! I tried using BluetoothA2DPSource to play some audio files stored in memory, but when I try to write a second audio data the program doesn't work, I tried to stop and reconnect and it also doesn't work. Any idea about how to do this?

Thanks in advance

**Comment by pschatzmann** [comment, marked as answer]
Just use the approach described in 'Sending Data from a A2DS Data Source with a Callback'.
In the callback you just provide the incremental data from your files. If you want to add some pause between the files just fill the buffer with 0 values.

**Reply by Hacktiv8or** [reply]
can you please provide some code or where to do this I have the same issue
my code--
`#include <Arduino.h>
#include "BluetoothA2DPSource.h"
#include "SD_MMC.h"
#include "FS.h"

BluetoothA2DPSource A2DP;
size_t current;
File file;

void connect() {
  A2DP.start("Bluetooth Music");
}
int32_t get_audio(Channels* dt, int32_t len) {
  size_t result_len_bytes = file.read((uint8_t*)dt, len * sizeof(int16_t) * 2);
  int32_t result_len = result_len_bytes / (sizeof(int16_t) * 2);
  // ESP_LOGD("get_sound_data", "%d -> %d",len );
  return result_len;
}

void setup() {
  SD_MMC.begin();
  file = SD_MMC.open("/asdf_main.raw");
  // connect();
  // A2DP.set_auto_reconnect(false);
  A2DP.start("Bluetooth Music", get_audio);
  delay(500);
  delay(2000);
  SD_MMC.begin();
  file = SD_MMC.open("/asdf.raw");
  A2DP.start("Bluetooth Music", get_audio);
  delay(500);
  delay(2000);
}

void loop() {
  delay(10); // to prevent watchdog (as per example)
}`
